Tests/SwiftNavigationTests/IsolationTests.swift

```diff
@@ -3,34 +3,38 @@
   import XCTest
 
   class IsolationTests: XCTestCase {
-    @MainActor
-    func testIsolationOnMinActor() async {
-      let model = MainActorModel()
-      let expectation = expectation(description: "observation")
-      expectation.expectedFulfillmentCount = 2
-      let token = SwiftNavigation.observe {
-        _ = model.count
-        MainActor.assertIsolated()
-        expectation.fulfill()
+    func testIsolationOnMainActor() async throws {
+      try await Task { @MainActor in
+        let model = MainActorModel()
+        var didObserve = false
+        let token = SwiftNavigation.observe {
+          _ = model.count
+          MainActor.assertIsolated()
+          didObserve = true
```

Comment:
Wasm also doesn't support test expectations (AFAICT), and so I've replaced them with more rudimentary tools for now.